pkg/router/template/router.go

```diff
@@ -385,7 +382,17 @@ func (r *templateRouter) writeDefaultCert() error {
 		} else {
 			r.defaultCertificatePath = outPath
 		}
-		if err := r.watchSecretDir(r.defaultCertificateDir, outPath); err != nil {
+		reloadFn := func() {
+			log.V(0).Info("updating default certificate", "path", outPath)
+			os.Remove(outPath)
```

Comment:
What happens if `os.Remove()` is successful but the subsequent write fails in `secretToPem`? Are we in a state where any other subsequent reload of the router will fail (because haproxy barfs with incomplete files and/or data)?

Reply:
Now that you mention it, we should improve this behavior.  That said, `writeDefaultCert` had the same issue before this change, so I don't think this makes matters worse than before.  What do you think?

Reply:
> Now that you mention it, we should improve this behavior. That said, `writeDefaultCert` had the same issue before this change, so I don't think this makes matters worse than before. What do you think?

In a straight run I think things will be generally fine. Just wanted to raise this; we can iterate post this PR merging.